admin/section/class-convertkit-admin-section-general.php

```diff
@@ -598,25 +598,68 @@ public function maybe_initialize_and_refresh_resources() {
 		}
 
 		// Refresh Forms.
-		$this->forms->refresh();
+		$result = $this->forms->refresh();
+
+		// Bail if an error occured.
+		if ( is_wp_error( $result ) ) {
+			// Delete credentials if the error is a 401.
+			convertkit_maybe_delete_credentials( $result, CONVERTKIT_OAUTH_CLIENT_ID );
+			return;
+		}
 
 		// Also refresh Landing Pages, Tags and Posts. Whilst not displayed in the Plugin Settings, this ensures up to date
 		// lists are stored for when editing e.g. Pages.
 		$landing_pages = new ConvertKit_Resource_Landing_Pages( 'settings' );
-		$landing_pages->refresh();
+		$result        = $landing_pages->refresh();
+
+		// Bail if an error occured.
+		if ( is_wp_error( $result ) ) {
+			// Delete credentials if the error is a 401.
+			convertkit_maybe_delete_credentials( $result, CONVERTKIT_OAUTH_CLIENT_ID );
```

Comment:
Can we just put this in the refresh methods at the highest level so we don't have to remember it check it and update all these locations?